### PR TITLE
add codecov report and badge

### PR DESCRIPTION
**Pull request by AABur**
_(related #232)_
Adds GHA to upload code coverage report to [Codecov](https://about.codecov.io/)
Adds Codecov badge to README.md

Для подключения Codecov необходимо:
1. Sign up on [Codecov](https://about.codecov.io/sign-up) and link GitHub
2. Install the [GitHub App Integration](https://github.com/apps/codecov).
3. Update badge https://docs.codecov.com/docs/status-badges

### CODE REVIEW

**Review by AABur — COMMENTED** [review]
test_machete.py - got here by mistake. I can't figure out how to get it out from PR

**Comment by PawelLipski** [comment]
Whoops `general checks` are spuriously failing on (supposedly) dead links... I'll fix that on `develop` soon

**Comment by AABur** [comment]
> Whoops `general checks` are spuriously failing on (supposedly) dead links... I'll fix that on `develop` soon

I checked based on this [report](https://app.circleci.com/pipelines/github/VirtusLab/git-machete/1856/workflows/b0d4250b-d5bb-4d9c-b614-21b74d802852/jobs/8318/parallel-runs/0/steps/0-103). All the links are alive. There was some kind of error in the `general checks`.

**Comment by PawelLipski** [comment]
> test_machete.py - got here by mistake. I can't figure out how to get it out from PR

`git checkout develop <path-to-file>`, commit and push

**Comment by AABur** [comment]
> > test_machete.py - got here by mistake. I can't figure out how to get it out from PR
> 
> `git checkout develop <path-to-file>`, commit and push

Thank you! DONE!

**Comment by PawelLipski** [comment]
> > Whoops `general checks` are spuriously failing on (supposedly) dead links... I'll fix that on `develop` soon
> 
> I checked based on this [report](https://app.circleci.com/pipelines/github/VirtusLab/git-machete/1856/workflows/b0d4250b-d5bb-4d9c-b614-21b74d802852/jobs/8318/parallel-runs/0/steps/0-103). All the links are alive. There was some kind of error in the `general checks`.

Pls rebase onto the most recent `develop`, should be fixed now

**Comment by PawelLipski** [comment]
I'm now trying to make Codecov actually see this repo... will get back to you soon, hopefully

File: .github/workflows/codecov.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    - name: Setup Python
+      uses: actions/setup-python@master
+      with:
+        python-version: 3.6
+    - name: Generate coverage report
+      run: |
+        pip install pytest-cov
+        pytest --cov --cov-report xml .
+    - name: Upload coverage to Codecov
+      uses: codecov/codecov-action@v2
```
Comment (PawelLipski) [inline .github/workflows/codecov.yaml]: Actually! for the sake of consistency, as I don't want to do yet another CI migration without a good reason... could you use CircleCI's `codecov/codecov` Orb and plug it into the existing pipeline in `.circleci/config.yml`? There's a ready sample in our another open source project, [VirtusLab-Open-Source/strapi-plugin-comments](https://github.com/VirtusLab-Open-Source/strapi-plugin-comments/blob/master/.circleci/config.yml)

Comment (PawelLipski) [reply .github/workflows/codecov.yaml]: This will also make it easier to actually test the coverage upload without setting up GH Actions on this repo 😅

Comment (PawelLipski) [reply .github/workflows/codecov.yaml]: To stay within the existing tooling, pls add `coverage xml` to `testenv:coverage`->`commands` within tox.ini; then you'll be able to simply upload the generated `coverage.xml` file to Codecov as a `codecov/upload` `step` under `test against python 3_9 and git 2_30_1 with coverage report` job in CircleCI config

Comment (AABur) [reply .github/workflows/codecov.yaml]: No problem. I'll add Cidecov integration to CicrleCi pipeline.

Comment (AABur) [reply .github/workflows/codecov.yaml]: DONE!!!
GitHub Action - removed
CircleCi - updated

Next steps: 
- [Getting a Codecov account](https://docs.codecov.com/docs/step-2-getting-a-codecov-account-and-uploading-coverage)
- fix [Codecov badge](https://docs.codecov.com/docs/status-badges)

Comment (PawelLipski) [reply .github/workflows/codecov.yaml]: Gr8 🎉 lemme see codecov.com now 🕵🏻

**Comment by PawelLipski** [comment]
Okay! lemme merge this PR, I might still need to massage this a bit but I'll take care of that myself.

Feel free to proceed with your original issue of choice (split test_machete.py), or whatever other yet-unassigned `good first issue` as you wish

**Review by PawelLipski — APPROVED** [review]

File: .github/workflows/codecov.yaml (hunk 1 above)
Comment (PawelLipski) [reply .github/workflows/codecov.yaml]: Seems to work:

https://app.codecov.io/gh/VirtusLab/git-machete/commits?page=1

Comment (PawelLipski) [reply .github/workflows/codecov.yaml]: > fix [Codecov badge](https://docs.codecov.com/docs/status-badges)

Badge doesn't need fixing, just a matter of getting these changes merged to master (will happen with the next release soon). Then the badge will show up in the main repo page's README, with correct percentage

Also, looks that the token attached to the codecov.io URL is redundant now, I'll remove it

**Pull request merged**